app/layouts/DashboardLayout/SetPinModal.view/SetPinModal.test.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom/extend-expect';
+import '../../../testUtils/i18nForTests';
+
+import { SetPinModal } from './SetPinModal.view';
+
+describe('SetPinModal', () => {
+  const setPinMsg =
+    'Your Transaction PIN is used as an extra safety measure. It is required if a transaction value is higher than the threshold MOB amount.';
+  test('when isShown prop is false the modal is not rendered', () => {
+    const handleOnClick = jest.fn();
+
+    render(<SetPinModal onPinSubmit={handleOnClick} isShown={false} />);
+
+    const pinModal = screen.queryByText(setPinMsg);
+    expect(pinModal).toBeNull();
+  });
+
+  test('render and submit with valid form data', async () => {
+    const handleOnClick = jest.fn();
+    const pin = '123456';
+    const pinThreshold = '10.000000000000';
+
+    render(<SetPinModal onPinSubmit={handleOnClick} isShown />);
+
+    const newPinField = screen.getByLabelText('PIN', {
+      selector: 'input',
+    }) as HTMLInputElement;
+    const confirmPinField = screen.getByLabelText('PIN Confirmation', {
+      selector: 'input',
+    }) as HTMLInputElement;
+    const pinThresholdField = screen.getByLabelText('Minimum amount that requires a PIN', {
+      selector: 'input',
+    }) as HTMLInputElement;
+    const submitButton = screen.getByRole('button', { name: 'Set PIN' });
+
+    userEvent.type(newPinField, pin);
+    userEvent.type(confirmPinField, pin);
+    userEvent.type(pinThresholdField, pinThreshold);
```

Comment:
Isn't act(...) needed here? Formik needs it.

Reply:
Nope, it works. On line 50 theres a test to throw an error for an invalid entry, which means it had to actually receive the input to know when to throw the error and when not.

Reply:
Weird...